sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -361,11 +364,34 @@ object ViewHelper {
     }
   }
 
+  /**
+   * Convert the view query SQL configs in `properties`.
+   */
+  private def generateQuerySQLConfigs(conf: SQLConf): Map[String, String] = {
+    val modifiedConfs = conf.getAllConfs.filter { case (k, _) =>
+      conf.isModifiable(k) && k != SQLConf.MAX_NESTED_VIEW_DEPTH.key
```

Comment:
IMHO, we should keep the captured SQL configs as less as possible. Because it's hard to precisely captures the configs which only affect the parser and analyzer, an alternative way is just filtering out the configs that definitely can NOT affect parsing/analyzing.